sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -477,7 +477,8 @@ object RemoveRedundantAliases extends Rule[LogicalPlan] {
 object RemoveNoopOperators extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     // Eliminate no-op Projects
-    case p @ Project(_, child) if child.sameOutput(p) => child
+    case Project(projList, child) if projList.length == child.output.length &&
+      projList.zip(child.output).forall { case (e1, e2) => e1.semanticEquals(e2) } => child
```

Comment:
Is this for Alias?

Reply:
yea, yes. I modified the code to handle the case below;
```
:- Project [v#183]
:  +- Project [cast(v#183 as decimal(11,0)) AS v#183]
:     +- Project [v#183]
```

Reply:
Can you a comment for this?

Reply:
I'm not sure when we add `RemoveNoopOperators` if we consider this case. But seems this change didn't cause any test failure.